File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      hm = {}
      res = []
      for i in range(len(nums)):
        hm[nums[i]] = i
      for j in range(len(nums)):
        com = target - nums[j]
        if com in hm and hm[com] != j:    
            res.append(min(hm[com], j))
            res.append(max(hm[com], j))
            break
      return res

File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int N = heights.length;
        int area = 0;
        for(int i = 0; i < N; i++) {
            for(int j = i +1; j < N; j++) {
                area = Math.max(area, ((j-i) * Math.min(heights[i], heights[j])));
            }

        }
        return area;
    }
}
